core/services/keeper/registry_synchronizer_process_logs.go

```diff
@@ -5,17 +5,17 @@ import (
 	"sync"
 
 	"github.com/smartcontractkit/chainlink/core/chains/evm/log"
-	"github.com/smartcontractkit/chainlink/core/internal/gethwrappers/generated/keeper_registry_wrapper1_1"
 	"github.com/smartcontractkit/chainlink/core/services/keystore/keys/ethkey"
 )
 
 func (rs *RegistrySynchronizer) processLogs() {
 	wg := sync.WaitGroup{}
-	wg.Add(4)
+	wg.Add(5)
 	go rs.handleSyncRegistryLog(wg.Done)
 	go rs.handleUpkeepCanceledLogs(wg.Done)
 	go rs.handleUpkeepRegisteredLogs(wg.Done)
 	go rs.handleUpkeepPerformedLogs(wg.Done)
+	go rs.handleUpkeepGasLimitSetLogs(wg.Done)
```

Comment:
@RyanRHall 's comments: Potentially think about separating rs.handleSyncRegistryLog & rest logs from being in the same wait group. Probably needs a different ticket for investigation

Reply:
38161 for investigation